lib/buffer.js

```diff
@@ -7,62 +7,219 @@ const alloc = smalloc.alloc;
 const truncate = smalloc.truncate;
 const sliceOnto = smalloc.sliceOnto;
 const kMaxLength = smalloc.kMaxLength;
+const useOldBuffer = !!process.useOldBuffer;
 
-exports.Buffer = Buffer;
-exports.SlowBuffer = SlowBuffer;
-exports.INSPECT_MAX_BYTES = 50;
+var Buffer, SlowBuffer, NativeBuffer;
 
 
-Buffer.poolSize = 8 * 1024;
-var poolSize, poolOffset, allocPool;
+// New Buffer implementation using Uint8Array. Done this way so the constructor
+// name can remain "Buffer" and "SlowBuffer".
 
+if (!useOldBuffer) {
+  Buffer = function Buffer(arg) {
+    // Common case.
+    if (typeof arg === 'number') {
+      // If less than zero, or NaN.
+      if (arg < 0 || !(arg <= 0 || arg > 0))
+        arg = 0;
+      return allocate(arg);
+    }
 
-function createPool() {
-  poolSize = Buffer.poolSize;
-  allocPool = alloc({}, poolSize);
-  poolOffset = 0;
+    // Slightly less common case.
+    if (typeof arg === 'string') {
+      return fromString(arg, arguments[1]);
+    }
+
+    // Unusual.
+    return fromObject(arg);
+  };
+
+  Buffer.prototype.__proto__ = Uint8Array.prototype;
+  Buffer.__proto__ = Uint8Array;
```

Comment:
Ah, so inheritance correctly works from Typed Arrays, just not `Array`? (Without needing `class`? I thought you said it did?)

I am both slightly disappointed and very glad that my terrible wrapper hack(s) didn't need to be used. :P